components/script/dom/readablebytestreamcontroller.rs

```diff
@@ -52,39 +182,857 @@ impl ReadableByteStreamController {
     ) -> Fallible<()> {
         todo!()
     }
+
+    /// <https://streams.spec.whatwg.org/#readable-byte-stream-controller-get-desired-size>
+    pub(crate) fn get_desired_size(&self) -> Option<f64> {
+        // Let state be controller.[[stream]].[[state]].
+        let stream = self.stream.get()?;
+
+        // If state is "errored", return null.
+        if stream.is_errored() {
+            return None;
+        }
+
+        // If state is "closed", return 0.
+        if stream.is_closed() {
+            return Some(0.0);
+        }
+
+        // Return controller.[[strategyHWM]] − controller.[[queueTotalSize]].
+        Some(self.strategy_hwm.get() - self.queue_total_size.get())
+    }
+
+    /// <https://streams.spec.whatwg.org/#abstract-opdef-readablebytestreamcontrollergetbyobrequest>
+    #[allow(unsafe_code)]
+    pub(crate) fn get_byob_request(
+        &self,
+        can_gc: CanGc,
+    ) -> Fallible<Option<DomRoot<ReadableStreamBYOBRequest>>> {
+        let cx = GlobalScope::get_cx();
+
+        // If controller.[[byobRequest]] is null and controller.[[pendingPullIntos]] is not empty,
+        let pending_pull_intos = self.pending_pull_intos.borrow();
```

Comment:
Same here, if you make this `mut`, can you remove the internal refcells on the descriptor?